### ISSUE TITLE
Sms Autofill for Android not Working for internal testing on Play console

### ISSUE THREAD

**Issue by smit-locusnine** [Tkko/Flutter_Pinput#108]
I tried using the package and autofill worked for me when I started the app on release mode. As soon as I uploaded to internal testing it didn't worked for me.

                          `Pinput(
                                  controller: _pinPutController,
                                  focusNode: _pinPutFocus,
                                  mainAxisAlignment:
                                      MainAxisAlignment.spaceBetween,
                                  showCursor: true,
                                  cursor: TextStyles.cursor,
                                  length: 4,
                                  inputFormatters: [FilteringTextInputFormatter
                                      .digitsOnly,],
                                  pinAnimationType: PinAnimationType.scale,
                                  androidSmsAutofillMethod:
                                      AndroidSmsAutofillMethod.smsRetrieverApi,
                                  defaultPinTheme: TextStyles.defaultPinTheme,
                                  focusedPinTheme: TextStyles.defaultPinTheme
                                      .copyWith(
                                          decoration: TextStyles
                                              .pinPutSelectedDecoration),
                                  errorPinTheme:
                                      TextStyles.defaultPinTheme.copyWith(
                                    decoration: BoxDecoration(
                                      color: AppColors.kMain,
                                      borderRadius: BorderRadius.circular(8),
                                    ),
                                  ),
                                  listenForMultipleSmsOnAndroid: true,
                                  onCompleted: (val) {
                                    codeValue = val;
                                  },
                              )`

**Comment by kashiflab** [comment]
You need to generate signature from release hash key

**Comment by bigyank** [comment]
I dont know if you have figured this out already, but you have to add your SHA key from app store to firebase console. This is required if you are using firebase service with play store.

**Comment by Tkko** [comment]
Hey guys, it has to be the issue regarding the app signature, how did you get the hash?

**Comment by ELTEGANI** [comment]
Hi @Tkko  thanks for the quick reply. I used flutter run --release to get the release signature

**Comment by ELTEGANI** [comment]
Hi @Tkko  i was using `smsRetrieverApi` i switched it to `smsUserConsentApi` and its working after i release in play store. note my sms didnt start with `<#>` im wondering if its the reason that smsRetrieverApi not worked.

**Comment by Tkko** [comment]
@ELTEGANI  `smsUserConsentApi`  doesn't require app signature, In the [official docs](https://developers.google.com/identity/sms-retriever/verify) they aren't using <#> syntax, but it's wart trying.

**Comment by Tkko** [comment]
@ELTEGANI smsUserConsentApi doesn't require app signature that's why it works.
Try generating the hash by following official [docs](https://developers.google.com/identity/sms-retriever/verify)

**Comment by ELTEGANI** [comment]
@Tkko  yeah i tried this [class](https://github.com/googlearchive/android-credentials/blob/master/sms-verification/android/app/src/main/java/com/google/samples/smartlock/sms_verify/AppSignatureHelper.java) from official docs and i got the same signature your plugin printed.

**Comment by Tkko** [comment]
@ELTEGANI could you share what SMS looks like?

**Comment by ELTEGANI** [comment]
@Tkko it looks like **Your verification code for the fitness subscription program is : 123456
xxxxxxxxxxx** i also checked the size its below 140b

**Comment by Tkko** [comment]
@ELTEGANI take a look at [this](https://www.ackee.agency/blog/sms-verification-with-retriever-api) and follow the guide of calculating the hash, they are doing extra step while getting the app signature so, it could work.

**Comment by Tkko** [comment]
Closing the issue for now, feel free to reopen it if you have the same problem.
